subprojects/architecture-test/src/test/java/org/gradle/architecture/test/ServiceScopeAnnotationValidationTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.architecture.test;
+
+import com.tngtech.archunit.base.DescribedPredicate;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+import org.gradle.internal.Factory;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+import javax.inject.Inject;
+
+import static com.tngtech.archunit.base.DescribedPredicate.not;
+import static com.tngtech.archunit.lang.conditions.ArchPredicates.are;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+import static org.gradle.architecture.test.ArchUnitFixture.freeze;
+
+@AnalyzeClasses(packages = "org.gradle")
+public class ServiceScopeAnnotationValidationTest {
+
+    private static final DescribedPredicate<JavaClass> factory_classes = new DescribedPredicate<JavaClass>("Factory class", "") {
+        @Override
+        public boolean test(JavaClass javaClass) {
+            return !javaClass.isAssignableTo(Factory.class);
+        }
+    };
```

Comment:
@wolfs, This was a naïve implementation from my side, as I realized that many factory classes do not implement `Factory<T>` (maybe another rule is cooking up there 🤔).

Is, in your opinion, acceptable to filter factory classes by doing the `endsWith("Factory")` predicate?

Reply:
I think only `Factory` itself should be excluded here, not subclasses of it, since for `Factory<T>`, the `T` should probably be annotated with Service scope - or maybe it shouldn't...

Reply:
I'm thinking about this...

I think it's too broad to say that every `T` in `Factory<T>` should be annotated with `@ServiceScope`.  
 What you inject is a factory, which might return a value that doesn't necessarily need to be a service.

For example, the `Integer` in a `RandomFactory<Integer>` shouldn't be annotated, as it's not a service - it's a simple, pure value. 
On the other hand, IMO, the `RandomFactory` should be marked as a service, as it's the class handled by the DI subsystem, and upcoming features like https://github.com/gradle/gradle/pull/26786 need those annotations to decide if the service is injectable or not.

Reply:
Yes, I agree. Though you can inject a `Factory<Service>`, and that will give you lazy access to a certain service, I think even when you didn't implement a concrete subclass. That is why I think we should only exclude `Factory`, but not subtypes of it.

Reply:
But then, we would have a violation for `Integer` (using the example above), as it doesn't implement `@ServiceScope`

Reply:
I'm missing the clear differentiators for `T` types in `Factory<T>`, which clearly identifies `T` as a service

Reply:
I am saying: Ignore the type parameter of `Factory` for now, and ignore all classes equal to `Factory`, but not subtypes.

Reply:
Ah, I see! Sorry for the misunderstanding...